SCons/Defaults.xml

```diff
@@ -30,12 +30,13 @@ A function used to produce variables like &cv-link-_CPPINCFLAGS;. It takes
 four or five
 arguments: a prefix to concatenate onto each element, a list of
 elements, a suffix to concatenate onto each element, an environment
-for variable interpolation, and an optional function that will be
-called to transform the list before concatenation.
+for variable interpolation, an optional function that will be
+called to transform the list before concatenation, affect_signature which will wrap non-empty returned value with
```

Comment:
wrap in <parameter>.  This doesn't even fully describe the default function named `_concat`, which as above now looks like:
```
def _concat(prefix, items_iter, suffix, env, f=lambda x: x, target=None, source=None, affect_signature=True):
```
i.e. no mention of target/source.

It might be useful if this entry included something like "This &consvar; is usually set to an internal function, but can also be set to a user-supplied function". Or not... leave that to you.

Reply:
re explanation that user can also set
An expert would know that, a new user might do the wrong thing...
So I think leaving that from being explicitly stated is probably wise.

Next push will have update to args list, please comment if there's a better way to format that..

Reply:
I'll propose a tweak in a bit.

Reply:
That may have been a hasty promise... it's really hard to describe conscisely but accurately in words a function that almost nobody needs to know the details of that takes from four to eight arguments... I wonder if it's better to say nothing at all about arguments (that is, even less than now) in line with your last comment -

Reply:
Okay, here's a proposal:
```
<summary>
    <para>
        Holds a concatenator function used to process an iterable of
        unadorned path-oriented items into the form used for issuing a command,
        such as the transformation of &cv-link-CPPPATH; into
        preprocessor includes in &cv-link-_CPPINCFLAGS;.
        By default an internal function is used.
        An example of the usage of this function:
    </para>

    <example_commands>
env['_CPPINCFLAGS'] = '${_concat(INCPREFIX, CPPPATH, INCSUFFIX, __env__, RDirs, TARGET, SOURCE, affect_signature=False)}'
    </example_commands>

    <para>
        The function will always be called with four
        required parameters, and may be called with up to four
        optional parameters, which can also be given as keyword arguments.
        The mandatory parameters are:
        a prefix to pepend to each element;
        the list of elements;
        a suffix to append to each element;
        and a &consenv; for variable interpolation.
        The optional (keyword) arguments are:
        a transformer function (<parameter>f</parameter>),
        if supplied will be called to transform the list before concatenation;
        a target (<parameter>target</parameter>) and
        a source (<parameter>source</parameter>) for passing
        to the transformer function,
        and a signature flag (<parameter>affect_signature</parameter>)
        which indicates whether to wrap non-empty returned values with
        <literal>$(</literal> and <literal>$)</literal>to indicate the contents
        should not affect the signature of the generated command line.
    </para>

    </summary>
```

Reply:
I think will is not accurate? it's should.. because user could do the wrong thing here?